src/input/composite_device/mod.rs

```diff
@@ -257,22 +267,28 @@ impl CompositeDevice {
     ) -> Result<(), Box<dyn Error>> {
         log::debug!("Starting composite device");
 
+        let dbus_path = self.dbus_path.clone();
+
         // Start all source devices
         self.run_source_devices().await?;
 
         // Keep track of all target devices
         for (path, target) in targets.iter() {
             if let Err(e) = target.set_composite_device(self.client()).await {
-                return Err(
-                    format!("Failed to set composite device for target device: {:?}", e).into(),
-                );
+                return Err(format!(
+                    "Failed to set composite device for target device: {}",
+                    e.to_string()
+                )
```

Comment:
If the error already implements the `Display` trait, then you can just include the variable in the format string to do the equivalent:
```suggestion
                return Err(format!(
                    "Failed to set composite device for target device: {e}"
                )
```